config/provider.yml

```diff
@@ -22,4 +22,8 @@ services:
   Invertus\dpdBaltics\Provider\ProductAvailabilityProvider:
     arguments:
       - '@language'
-      - '@Invertus\dpdBaltics\Repository\ProductAvailabilityRepository'
\ No newline at end of file
+      - '@Invertus\dpdBaltics\Repository\ProductAvailabilityRepository'
+
+  Invertus\dpdBaltics\Provider\CurrentCountryProvider:
+    arguments:
+      - '@context'
```

Comment:
99% of the cases you dont need to inject whole context - just a FYI - you can create adapter and inject what you need from context - its easier to test stuff